### DISCUSSION TITLE
Determine the direction

### DISCUSSION THREAD

**Discussion by IvanIsak2000** [Spaceblock/cosmoterm#5] [Ideas]
Determine the direction of the project: is it a blockchain system, a social network using cryptography or another

**Comment by mb6ockatf** [comment]
what database management system we're going to use?

**Reply by IvanIsak2000** [reply]
I think SQlite or PostgreSQL.

The first is more familiar to me, but the second is often used in companies

**Reply by mb6ockatf** [reply]
ok, PostgreSQL then. I'm quite really familiar with it.
What about SQLAlchemy? It's widely used, but I dread using it; it's such a weird thing IMHO

**Reply by IvanIsak2000** [reply]
Yes, I also agree with using POstgreSQL

**Comment by mb6ockatf** [comment]
> Determine the direction of the project: is it a blockchain system, a social network using cryptography or another

let it be social networking core with great api, so everyone can develop his own interface (CLI, web or gui) with it?

**Reply by IvanIsak2000** [reply]
Let's say it will be a social network, but how can we take it with the theme of space?)

**Reply by mb6ockatf** [reply]
> Let's say it will be a social network, but how can we take it with the theme of space?)

yep... we'll have to change the name then - and inventing names is a tough thing, at least for me

**Reply by IvanIsak2000** [reply]
> yep... we'll have to change the name then - and inventing names is a tough thing, at least for me

The name of the organization / project is one of the primary tasks

**Reply by mb6ockatf** [reply]
> > yep... we'll have to change the name then - and inventing names is a tough thing, at least for me
> 
> The name of the organization / project is one of the primary tasks

if something connected with network - then, it can be something like `****net` - we need to invent the beginning of the word

**Reply by IvanIsak2000** [reply]
> > > yep... we'll have to change the name then - and inventing names is a tough thing, at least for me
> > 
> > 
> > The name of the organization / project is one of the primary tasks
> 
> if something connected with network - then, it can be something like `****net` - we need to invent the beginning of the word

CosmoNet? 🤔

**Reply by mb6ockatf** [reply]
idk. i'll think about name (hope you'll think, too)

**Comment by mb6ockatf** [comment]
If we're using PostgreSQL, we're going to handle some configuration. How we're going to do it? i have some ideas:
- default python `configparser`, `*.ini` file
- `*.toml` file - popular these days
- `*.yaml` - also popular, but it's syntax really sucks
- pass parameters with `argparse` python module
- get all args from environment variables

in the last 2 options it'll be possible to create neat shell script for starting out server. @IvanIsak2000 what do you think?

**Reply by IvanIsak2000** [reply]
These are new formats for me, would you like to use `.json`?)

**Reply by mb6ockatf** [reply]
> These are new formats for me, would you like to use `.json`?)

no, because it reminds me of javascript

**Comment by IvanIsak2000** [comment]
I agree that it is a very difficult task to implement Blockchain technology, but it means that you need to come up with an alternative

**Reply by IvanIsak2000** [reply]
> I agree that it is a very difficult task to implement Blockchain technology, but it means that you need to come up with an alternative

OK, I came up with a simple solution: at the initial stage, just create a space-themed social network

**Reply by IvanIsak2000** [reply]
Possible titles:
- Starnet
- Astronet
- Spacenet
- SkyNetwork

**Reply by mb6ockatf** [reply]
> > I agree that it is a very difficult task to implement Blockchain technology, but it means that you need to come up with an alternative
> 
> OK, I came up with a simple solution: at the initial stage, just create a space-themed social network

space-themed - it's not enough to become popular. we need an original idea

**Reply by mb6ockatf** [reply]
> Possible titles:
> 
> * Starnet
> * Astronet
> * Spacenet
> * SkyNetwork

i tried googling some of these names - there are already organisations with these names.

**Reply by mb6ockatf** [reply]
what about `termnetspace` (TNS)? like, social networking service that can be used out of CLI. and name connected with space topic

@IvanIsak2000 what do you think?

**Reply by IvanIsak2000** [reply]
> what about `termnetspace` (TNS)? like, social networking service that can be used out of CLI. and name connected with space topic
> 
> @IvanIsak2000 what do you think?

Not bad I like that the user will interact with the account/other people via the command line

**Reply by IvanIsak2000** [reply]
> > what about `termnetspace` (TNS)? like, social networking service that can be used out of CLI. and name connected with space topic
> > @IvanIsak2000 what do you think?
> 
> Not bad I like that the user will interact with the account/other people via the command line

This idea is quite cool, it will probably be easier to implement it, as well as to make widgets, downloads in the space theme

**Reply by IvanIsak2000** [reply]
@mb6ockatf 
That is, you want the user to write his message through the command line? Then you need to figure out how the terminal / program will send data to the server

**Reply by mb6ockatf** [reply]
api?

**Reply by IvanIsak2000** [reply]
Then we can change the name to `Astro termanal`

**Reply by mb6ockatf** [reply]
> Then we can change the name to `Astro termanal`

may be astterm?

**Reply by mb6ockatf** [reply]
then we need 2 repositories: 1 for server, other - for client application.
This one is going to contain the server part, i suppose

**Reply by IvanIsak2000** [reply]
> then we need 2 repositories: 1 for server, other - for client application. This one is going to contain the server part, i suppose

We need:
1) website - so that the user can register an account and get a token
2) a terminal application capable of sending requests

In my head it looks like
![image](https://user-images.githubusercontent.com/79650307/235462095-6c874546-cf5e-4be8-9ff3-ce457ce8006c.png)

**Reply by mb6ockatf** [reply]
yes! btw we don't have to include username in request, token is enough

**Reply by IvanIsak2000** [reply]
It seems to me that I came up with the concept of the project: look, there is a user who sends a message, the server accepts it and then forwards it to the recipient, while the message text is not recorded on the server, but only the hash is stored. Text and hashed data are stored only by the sender and recipient - hence the decentralized system 👀

Possible project names:
- FreeSpace - supposedly a free and protected place

**Reply by mb6ockatf** [reply]
1) that's not a decentralised system, since we still need a server. decentralised means the client is a server at the same time.
2) what for would we store message hash on the server?

i think i have a better idea. we can create a client that is a server at the same time. that would be a truly decentralised server.

if a recipient is currently offline (message failed to be sent), than it is stored offline at the sender's machine. when recipient is online, it sends a ping to all it's contacts, and if there are messages queried, than they're transmitted

name is great, i like it. let's make it lowerspace, as all linux programs names: freespace (like bspwm, vim, systemd - look, they're all lowercase)

**Reply by mb6ockatf** [reply]
that's a messager then. after we create this, we can make a kind of wall, where every member of this *wall* can write. messages of the wall are cached on every member's machine, and new ones are loaded when both machines are online

**Reply by mb6ockatf** [reply]
that would be a truly decentralised thing...

**Reply by mb6ockatf** [reply]
btw we can store wall and personal messages in a local blockchain, and then they're synced with all member's blockchains when they're online

**Reply by mb6ockatf** [reply]
then, personal chat is just a wall with 2 members

let's call it a wall, not chat?

**Reply by IvanIsak2000** [reply]
> then, personal chat is just a wall with 2 members
> 
> let's call it a wall, not chat?

maybe

**Comment by IvanIsak2000** [comment]
Hi!
What you think about `Cosmoterm` ?

`Cosmoterm` - secure terminal messenger

- We can make an ASCII picture on the welcome window

- A person can register / log in to an account both through the terminal and through the site

**Reply by IvanIsak2000** [reply]
> Hi! What you think about `Cosmoterm` ?
> 
> `Cosmoterm` - secure terminal messenger
> 
> * We can make an ASCII picture on the welcome window
> * A person can register / log in to an account both through the terminal and through the site

I did this 😊

![image](https://user-images.githubusercontent.com/79650307/236520664-fe37a87e-e11e-4fd5-9d12-42cd7e5ec306.png)

**Reply by IvanIsak2000** [reply]
How will we distribute `client` ? I suggest in `client.exe` extension

This is how it will look like:
![image](https://user-images.githubusercontent.com/79650307/236610853-388432f9-2c51-4231-b184-680c27886427.png)

**Reply by IvanIsak2000** [reply]
> How will we distribute `client` ? I suggest in `client.exe` extension
> 
> This is how it will look like: ![image](https://user-images.githubusercontent.com/79650307/236610853-388432f9-2c51-4231-b184-680c27886427.png)

Account window
![image](https://user-images.githubusercontent.com/79650307/236612133-f0012329-7294-4687-9704-bb0cc3473491.png)
